File: 1993-sum-of-all-subset-xor-totals/sum-of-all-subset-xor-totals.cpp
class Solution {
public:
    int subsetXORSum(vector<int>& nums) {
        int sz = nums.size(), ans = 0;
        for(int i = 0; i < (1 << sz); ++i){
            int currXor = 0;
            for(int j = 0; j < sz; ++j)
                if(i & (1 << j))
                    currXor ^= nums[j];
            ans += currXor;
        }
        return ans;
    }
};

File: 1993-sum-of-all-subset-xor-totals/sum-of-all-subset-xor-totals.py
class Solution(object):
    def subsetXORSum(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        size, ans = len(nums), 0
        for i in range(0, 1 << size):
            sub_ans = 0
            for j in range(0, size):
                if i & (1 << j):
                    sub_ans ^= nums[j]
            ans += sub_ans
        return ans
        